pkg/utils/cloudconfig/azure/config_test.go

```diff
@@ -0,0 +1,383 @@
+		},
+		"Location empty": {
+			config: &CloudConfig{
+				ARMClientConfig: azclient.ARMClientConfig{
+					Cloud: "c",
+				},
+				AzureAuthConfig: azclient.AzureAuthConfig{
+					UseManagedIdentityExtension: true,
+					UserAssignedIdentityID:      "a",
+				},
+				Location:             "",
+				SubscriptionID:       "s",
+				ResourceGroup:        "v",
+				ClusterName:          "c",
+				ClusterResourceGroup: "g",
+				VnetName:             "vn",
+			},
+			expectPass: false,
+		},
+		"ClusterName empty": {
+			config: &CloudConfig{
+				ARMClientConfig: azclient.ARMClientConfig{
+					Cloud: "c",
+				},
+				AzureAuthConfig: azclient.AzureAuthConfig{
+					UseManagedIdentityExtension: true,
+					UserAssignedIdentityID:      "a",
+				},
+				Location:             "l",
+				SubscriptionID:       "s",
+				ResourceGroup:        "v",
+				ClusterName:          "",
+				ClusterResourceGroup: "g",
+				VnetName:             "vn",
+			},
+			expectPass: false,
+		},
+		"ClusterResourceGroup empty": {
+			config: &CloudConfig{
+				ARMClientConfig: azclient.ARMClientConfig{
+					Cloud: "c",
+				},
+				AzureAuthConfig: azclient.AzureAuthConfig{
+					UseManagedIdentityExtension: true,
+					UserAssignedIdentityID:      "a",
+				},
+				Location:             "l",
+				SubscriptionID:       "s",
+				ResourceGroup:        "v",
+				ClusterName:          "c",
+				ClusterResourceGroup: "",
+				VnetName:             "vn",
+			},
+			expectPass: false,
+		},
+		"SubscriptionID empty": {
+			config: &CloudConfig{
+				ARMClientConfig: azclient.ARMClientConfig{
+					Cloud: "c",
+				},
+				AzureAuthConfig: azclient.AzureAuthConfig{
+					UseManagedIdentityExtension: true,
+					UserAssignedIdentityID:      "a",
+				},
+				Location:             "l",
+				SubscriptionID:       "",
+				ResourceGroup:        "v",
+				ClusterName:          "c",
+				ClusterResourceGroup: "g",
+				VnetName:             "vn",
+			},
+			expectPass: false,
+		},
+		"ResourceGroup empty": {
+			config: &CloudConfig{
+				ARMClientConfig: azclient.ARMClientConfig{
+					Cloud: "c",
+				},
+				AzureAuthConfig: azclient.AzureAuthConfig{
+					UseManagedIdentityExtension: true,
+					UserAssignedIdentityID:      "a",
+				},
+				Location:             "l",
+				SubscriptionID:       "s",
+				ResourceGroup:        "",
+				ClusterName:          "c",
+				ClusterResourceGroup: "g",
+				VnetName:             "vn",
+			},
+			expectPass: false,
+		},
+		"VnetName empty": {
+			config: &CloudConfig{
+				ARMClientConfig: azclient.ARMClientConfig{
+					Cloud: "c",
+				},
+				AzureAuthConfig: azclient.AzureAuthConfig{
+					UseManagedIdentityExtension: true,
+					UserAssignedIdentityID:      "a",
+				},
+				Location:             "l",
+				SubscriptionID:       "s",
+				ResourceGroup:        "v",
+				ClusterName:          "c",
+				ClusterResourceGroup: "g",
+				VnetName:             "",
+			},
+			expectPass: false,
+		},
+		"VnetResourceGroup empty": {
```

Comment:
In case of `VnetResourceGroup` is empty, please check if it's set to `resourceGroup`.

Reply:
fixed